data/src/main/java/com/linkedin/data/schema/SchemaToPdlEncoder.java

```diff
@@ -441,24 +502,29 @@ private void writeTyperef(TyperefDataSchema schema) throws IOException
         .writeSpace();
     DataSchema ref = schema.getRef();
     writeReferenceOrInline(ref, schema.isRefDeclaredInline());
+    recordSchemaElementLocation(schema);
   }
 
   private void writeMap(MapDataSchema schema) throws IOException
   {
+    markSchemaElementStartLocation();
     writeProperties(schema.getProperties());
     _builder.write("map[string")
         .writeComma()
         .writeSpace();
     writeReferenceOrInline(schema.getValues(), schema.isValuesDeclaredInline());
     _builder.write("]");
+    recordSchemaElementLocation(schema);
   }
 
   private void writeArray(ArrayDataSchema schema) throws IOException
   {
+    markSchemaElementStartLocation();
```

Comment:
instead adding these functions in every type write method can move it the switch case in encode and writeRecord field iterator. what is the purpose of it in each type?

Reply:
We don't call them for every type in the switch statement in encode. They aren't called for primitive types. Can certainly move this to encode but will not be as clean as just adding the two lines there since would have to call them only for non-primitives. Let me know what you prefer.